src/interaction/command/circle.ts

```diff
@@ -0,0 +1,260 @@
+      subcommand.addUserOption((option) => {
+        option.setName("owner");
+        option.setDescription("The owner of the circle (mention them)");
+        option.setRequired(true);
+        return option;
+      });
+      return subcommand;
+    });
+
+    // Adding "create channel" subcommand
+    this.slashCommand.addSubcommand((subcommand) => {
+      subcommand.setName("create-channel");
+      subcommand.setDescription("Create a channel for a circle");
+
+      // Set up all the options for the subcommand
+      // Channel name, and circle to add them to
+      subcommand.addStringOption((option) => {
+        option.setName("name");
+        option.setDescription("The name of the channel");
+        option.setRequired(true);
+        return option;
+      });
+      subcommand.addRoleOption((option) => {
+        option.setName("circle");
+        option.setDescription(
+          "The circle to create the channel for (reference the role)"
+        );
+        option.setRequired(true);
+        return option;
+      });
+
+      return subcommand;
+    });
+    this.slashCommand.addSubcommand((subcommand) => {
+      subcommand.setName("repost");
+      subcommand.setDescription("Repost the circle embeds");
+      return subcommand;
+    });
+  }
+
+  public async handleInteraction({ bot, interaction }: SlashCommandContext) {
+    const subComm = interaction.options.getSubcommand();
+    switch (subComm) {
+      case "add":
+        await interaction.deferReply();
+        await addCircle(bot, interaction);
+        break;
+      case "create-channel":
+        await interaction.deferReply();
+        await createChannel(bot, interaction);
+        break;
+      case "repost":
+        await bot.managers.circle.repost();
+        await interaction.reply("Updated circle data");
+    }
+  }
+}
+
+async function createChannel(bot: Bot, interaction: CommandInteraction) {
+  const name = interaction.options.getString("name", true);
+  const circleId = interaction.options.getRole("circle", true).id;
+  const guild = interaction.guild!;
+
+  const circle = bot.managers.database.cache.circles.get(
+    circleId
+  ) as CircleData;
+  if (!circle) {
+    await interaction.editReply({
+      content: "That circle does not exist.",
+    });
+    return;
+  }
+  const channelName = `${circle.emoji} ${name}`;
+  const channel = await guild.channels.create(channelName, {
+    type: "GUILD_TEXT",
+    parent: settings.circles.parentCategory,
+    permissionOverwrites: [
+      {
+        id: guild.roles.everyone,
+        deny: ["VIEW_CHANNEL"],
+        type: "role",
+      },
+      {
+        id: circleId,
+        allow: ["VIEW_CHANNEL"],
+        type: "role",
+      },
+    ],
+  });
+  const newSub = [...circle.subChannels, channel.id];
+  const res = await bot.managers.database.circleUpdate(circleId, {
+    subChannels: newSub,
+  });
+
+  if (!res) {
+    await interaction.editReply({
+      content: "Failed to update database",
```

Comment:
nit: would inform user that their channel was created but database could not be updated. Or, do something like line 222 and delete the new channel